src/systems/systems/src/navigation/radio_altimeter.rs

```diff
@@ -0,0 +1,864 @@
+            return;
+        }
+
+        // Update only once per refersh interval
+        if self.stale_for >= Self::ARINC_429_REFRESH_INTERVAL {
+            // Interrogate the transceivers for a response (which might fail if no physical path exists
+            // or some other kind of failure).
+            if let Some(response) = transceivers.response() {
+                let travel_time = response.travel_time();
+
+                let speed: Velocity = Velocity::new::<meter_per_second>(299_792_458.)
+                    * Ratio::new::<ratio>(0.999707085);
+                let traveled_distance: Length = travel_time * speed;
+                self.radio_altitude = (traveled_distance
+                    - self.aircraft_installation_delay.delay())
+                    / Ratio::new::<ratio>(2.);
+                self.is_tracking_ground = true;
+
+                // Instead of setting it to Duration::ZERO, subtract to ensure that we average out
+                // the correct information rate over multiple refreshes. This means the next update
+                // might come a little sooner. We also use a loop so that we don't fall behind if
+                // the frame rate spikes very high and we get an enormous delta.
+                while self.stale_for >= Self::ARINC_429_REFRESH_INTERVAL {
+                    self.stale_for -= Self::ARINC_429_REFRESH_INTERVAL;
+                }
+            } else {
+                // We lost the ground target
+                self.is_tracking_ground = false;
+            }
+        } else {
+            self.stale_for += context.delta();
+        }
+    }
+
+    fn radio_altitude(&self) -> Length {
+        // round to nearest 0.125
+        (self.radio_altitude * Ratio::new::<ratio>(8.)).round::<foot>() / Ratio::new::<ratio>(8.)
+    }
+
+    fn ssm(&self) -> SignStatus {
+        if self.remaining_self_check > Duration::ZERO {
+            SignStatus::FailureWarning
+        } else if !self.is_tracking_ground {
+            SignStatus::NoComputedData
+        } else {
+            SignStatus::NormalOperation
+        }
+    }
+}
+
+pub struct Ala52BRadioAltimeter {
+    // Pins
+    number: usize,
+    aircraft_installation_delay: Ala52BAircraftInstallationDelay,
+
+    // failure
+    failure: Failure,
+
+    // Power
+    powered_by: ElectricalBusType,
+    is_powered: bool,
+
+    /// How long the Radio Altimeter can tolerate a power loss and continue functioning.
+    power_holdover: Duration,
+    unpowered_for: Duration,
+
+    /// How long the self checks take for runtimes running on this Radio Altimeter.
+    self_check_time: Duration,
+
+    runtime: Option<Ala52BRadioAltimeterRuntime>,
+
+    // Outputs
+    radio_altitude_id: VariableIdentifier,
+}
+
+impl Ala52BRadioAltimeter {
+    const MINIMUM_STARTUP_TIME_MILLIS: u64 = 4_000;
+    const MAXIMUM_STARTUP_TIME_MILLIS: u64 = 6_000;
+    const MINIMUM_POWER_HOLDOVER: u64 = 30_000;
+    const MAXIMUM_POWER_HOLDOVER: u64 = 35_000;
```

Comment:
It's interesting to see that it has such a long time it can stay "alive" without power!